### PR TITLE
Refactor long

### PR DESCRIPTION
**Pull request by haraschax**
- Replace speed_smoother with MPC, this makes cruise accel and deccel smoother while increasing limits.
- Change longitudinal planner output API to canonical form used in lateral planner
- General cleanup
- Get planner ready for future longitudinal improvements

### CODE REVIEW

File: selfdrive/test/test_onroad.py [hunk 1]
```diff
@@ -154,7 +154,7 @@ def test_cpu_usage(self):
     self.assertTrue(cpu_ok)
 
   def test_model_timings(self):
-    cfgs = [("modelV2", 0.035, 0.03), ("driverState", 0.022, 0.020)]
+    cfgs = [("modelV2", 0.035, 0.03), ("driverState", 0.025, 0.021)]
```
Comment (adeebshihadeh) [inline selfdrive/test/test_onroad.py:157]: why did this go up?

Comment (HaraldSchafer) [reply selfdrive/test/test_onroad.py:157]: Not sure? I assumed it was stochastic and just failed randomly.

File: selfdrive/controls/lib/longitudinal_mpc_lib/SConscript [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    'lib_mpc_export/acado_common.h',
+    'lib_mpc_export/acado_auxiliary_functions.h',
+    'lib_mpc_export/acado_qpoases_interface.hpp',
+]
+
+interface_dir = Dir('lib_mpc_export')
+
+SConscript(['#phonelibs/qpoases/SConscript'], variant_dir='lib_qp', exports=['interface_dir'])
+
+if GetOption('mpc_generate'):
```
Comment (adeebshihadeh) [inline selfdrive/controls/lib/longitudinal_mpc_lib/SConscript:32]: wrong indendation

**Comment by kennedyshead** [comment]
```
crash
Traceback (most recent call last):
  File "capnp/lib/capnp.pyx", line 1250, in capnp.lib.capnp._DynamicStructBuilder.__getattr__
  File "capnp/lib/capnp.pyx", line 1243, in capnp.lib.capnp._DynamicStructBuilder._get
capnp.lib.capnp.KjException: capnp/schema.c++:493: failed: struct has no such member; name = speeds

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/data/openpilot/selfdrive/manager/process.py", line 36, in launcher
    mod.main()
  File "/data/openpilot/selfdrive/controls/controlsd.py", line 673, in main
    controls.controlsd_thread()
  File "/data/openpilot/selfdrive/controls/controlsd.py", line 667, in controlsd_thread
    self.step()
  File "/data/openpilot/selfdrive/controls/controlsd.py", line 657, in step
    actuators, lac_log = self.state_control(CS)
  File "/data/openpilot/selfdrive/controls/controlsd.py", line 455, in state_control
    actuators.gas, actuators.brake, self.v_target, self.a_target = self.LoC.update(self.active, CS, self.CP, long_plan)
  File "/data/openpilot/selfdrive/controls/lib/longcontrol.py", line 76, in update
    if len(long_plan.speeds) == CONTROL_N:
  File "capnp/lib/capnp.pyx", line 1252, in capnp.lib.capnp._DynamicStructBuilder.__getattr__
  File "capnp/lib/capnp.pyx", line 1250, in capnp.lib.capnp._DynamicStructBuilder.__getattr__
  File "capnp/lib/capnp.pyx", line 1243, in capnp.lib.capnp._DynamicStructBuilder._get
AttributeError: capnp/schema.c++:493: failed: struct has no such member; name = speeds
Process controlsd:
Traceback (most recent call last):
  File "capnp/lib/capnp.pyx", line 1250, in capnp.lib.capnp._DynamicStructBuilder.__getattr__
  File "capnp/lib/capnp.pyx", line 1243, in capnp.lib.capnp._DynamicStructBuilder._get
capnp.lib.capnp.KjException: capnp/schema.c++:493: failed: struct has no such member; name = speeds
```

**Pull request merged**